sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -307,6 +307,7 @@ case class MapPartitions[T, U](
     uEncoder: ExpressionEncoder[U],
     output: Seq[Attribute],
     child: SparkPlan) extends UnaryNode {
+  override def missingInput: AttributeSet = AttributeSet.empty
```

Comment:
This fixes the problem, but I think it might be better to add something like `def producedAttributes: AttributeSet = AttributeSet` to `QueryPlan`.  This can be automatically subtracted from `missingInput` and nodes that produce attributes can override this function.  I think that will be clearer.  What do you think?